### PR TITLE
Added gitignore defs, editorconfig and hookspecs representation

### PR DESCRIPTION
**Pull request by TanyaEleventhGoddess**

### CODE REVIEW

**Comment by TanyaEleventhGoddess** [comment]
Side note, I have another patch which introduces proper kwargs support in adv_testing repo on my fork
However, this depends on this patch adoption as reverting to the previous state would've broken a bit my editor

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,75 @@
+## Linux ##
```
Comment (RonnyPfannschmidt) [inline .gitignore:1]: i'm starting to think there should be global gitignores people are advised to use (its a pain to have all the repos ignore all the folders of all the tools)

File: src/pluggy/hooks.py [hunk 2]
```diff
@@ -240,7 +240,13 @@ def _add_hookimpl(self, hookimpl):
             methods.insert(i + 1, hookimpl)
 
     def __repr__(self):
-        return "<_HookCaller %r>" % (self.name,)
+        return "<_HookCaller %r, _wrappers=%r, _nonwrappers=%r, _call_history=%r, spec=%r>" % (
```
Comment (RonnyPfannschmidt) [inline src/pluggy/hooks.py:243]: i would like to see a example on the effect this has when more than just one plugin is added to a setup

i believe its going to be a problem as proposed here

**Comment by TanyaEleventhGoddess** [comment]
Sun, 11 Jul 2021 08:06:21 +0000 (UTC) Ronny Pfannschmidt ***@***.***>:
> @RonnyPfannschmidt commented on this pull request.
> 
> 
> 
> > @@ -1,3 +1,75 @@
> +## Linux ##
> 
> i'm starting to think there should be global gitignores people are advised to use (its a pain to have all the repos ignore all the folders of all the tools)
> 

You're right, but that's up to git mantainers to do. As simple config files to copy, there is a database already at https://github.com/github/gitignore

File: src/pluggy/hooks.py (hunk 2 above)
Comment (TanyaEleventhGoddess) [reply src/pluggy/hooks.py]: See if the new version fits your liking

File: .gitignore (hunk 1 above)
Comment (goodboy) [reply .gitignore:1]: Agreed.

I don't really see how this has anything to do with this repo personally; users should be more then capable of managing their own 😂

**Comment by goodboy** [comment]
@TanyaEleventhGoddess can you post some output from this changeset and show the use case?

I can see wanting more details while debugging from console but i'm wondering if there's some `exec`-able properties  of `__repr__` we might want to keep in place.

File: .gitignore (hunk 1 above)
Comment (TanyaEleventhGoddess) [reply .gitignore:1]: Honestly I've seen many repos including them (even urllib3 if I recall correctly) so I thought I should've add it

**Comment by TanyaEleventhGoddess** [comment]
Mon, 16 Aug 2021 06:51:42 -0700 goodboy ***@***.***>:
> @TanyaEleventhGoddess can you post some output from this changeset and show the use case?
> 
> I can see wanting more details while debugging from console but i'm wondering if there's some `exec`-able properties  of `__repr__` we might want to keep in place. 
> 
What do you mean as "output from this changeset"? Anyway using `__repr__` for displaying data structures in form of string is the recommended [solution](https://www.journaldev.com/22460/python-str-repr-functions)

**Comment by goodboy** [comment]
Are we at some kind of stop gap on this @RonnyPfannschmidt ?

**Comment by RonnyPfannschmidt** [comment]
i have been stripped of time a bit as of a while now,

my primary issues with the pr are that it 

a) adds untested code 
b) pushes a new configfile and dozens of additional gitignores (that really should be global)

**Pull request closed without merging**